obspy/io/rg16/util.py

```diff
@@ -94,7 +94,8 @@ def _read(fi, position, length, dtype):
     if dtype in READ_FUNCS:
         return READ_FUNCS[dtype](fi, length)
     else:
-        data = np.fromstring(fi.read(int(length)), dtype)
+        # multiply 1 to get new array with type np.float32 see #2198
+        data = np.fromstring(fi.read(int(length)), dtype).astype(np.float32)
```

Comment:
Ah, that looks much better. :tada: 
Now if you fix that comment above it I'm happy.. ;-)